### DISCUSSION TITLE
How to Remove Logging Output from Allure 2?

### DISCUSSION THREAD

**Discussion by iriiiina** [allure-framework/allure2#1846] [Questions & Support] [answered]
I have Java/Groovy/Spock tests and Allure 2 report. I use allure.Step() annotation to describe all steps that I want to see in Allure report. For example:

```
def "some test"() {
  when: "do some request"
  Request.getRequest()
}

class Request {
  @Step("GET https://some.request")
  static Response getRequest() {
    return given().log().all()
            .filter(new AllureRestAssured())
            .when()
            .get(conf.url())
  }
}
```
In that case I see two basically same blocks in report:
1. first beautiful block step with finely formatted request and response due to `AllureRestAssured()` - I want to leave it
2. second unformatted block with the same request and response info due to `.log().all()` console output - I want to remove this one

<img width="752" alt="screenshot" src="https://user-images.githubusercontent.com/5726961/204296084-a1769b80-b0d5-4759-9777-bdd9091b4a47.png">

Is there a way to remove all console output from Allure report and leave only methods that have `@Step` annotation?

I have seen the other discussion #1446 where is mentioned WDIO config `addConsoleLogs`, but I have't found something like this for Spock or Java.

**Comment by Guter1988** [comment]
It looks like at Allure, we have all the output logs attached as steps
`    public void standardOut(String message) {
        this.logMessage(message, Status.PASSED);
    }

    public void standardErr(String message) {
        this.logMessage(message, Status.BROKEN);
    }

    private void logMessage(String message, Status status) {
        if (!Objects.isNull(message)) {
            String trimmed = message.trim();
            if (!trimmed.isEmpty()) {
                this.getLifecycle().getCurrentTestCaseOrStep().ifPresent((parentUuid) -> {
                    String uuid = UUID.randomUUID().toString();
                    this.getLifecycle().startStep(parentUuid, uuid, (new StepResult()).setName(trimmed).setStatus(status));
                    this.getLifecycle().stopStep(uuid);
                });
            }
        }
    }`

So any log at the stdout will be added.

We have the same issue.

**Comment by iriiiina** [comment, marked as answer]
Ma college have found a solution. Separate object with all required RestAssured filters:
```
class BeforeAll {
    BeforeAll() {
        filters(new AllureRestAssured(), new RequestLoggingFilter(), new ResponseLoggingFilter())
    }
}
```
And call this object on test or extended Specification:
```
static BeforeAll beforeAll = new BeforeAll()
````
